modules/ximgproc/src/radon_transform.cpp

```diff
@@ -0,0 +1,194 @@
+  /**
+   * A utility function for generating a vector of linearly spaced integers.
+   */
+
+  // Finding diff
+  float diff = (end - start)*1.0f/(n-1);
+  std::vector<float> vec(n);
+  float val = start;
+
+  for(int i=0; i<n; ++i) {
+    vec[i] = val;
+    // Adding diff to val
+    val += diff;
+  }
+
+  // Returning the vector
+  return vec;
+}
+
+/**
+ * [meshgrid Returns a rectangular grid 2D space]
+ * @param matX [Repeating vector along X direction]
+ * @param matY [Repeating vector along Y direction]
+ * @param X    [Final meshgrid matrix X]
+ * @param Y    [Final meshgrid matrix Y]
+ */
+static inline void meshgrid(const Mat &matX, const Mat &matY, Mat &X, Mat &Y) {
+  /**
+   * A utitlity function to compute the meshgrid matrices for given input vectors.
+   */
+  repeat(matX.reshape(1,1), static_cast<int>(matY.total()), 1, X);
+  repeat(matY.reshape(1,1).t(), 1, static_cast<int>(matX.total()), Y);
+}
+
+/**
+ * [radonTransform Computes the Radon Transform of the given input array]
+ * @param src      [Input image]
+ * @param dst      [Output sinogram, referenced]
+ * @param accuracy [Minimum accuracy for angles ( IN DEGREES ), defaults to 1 degree]
+ */
+void radonTransform(InputArray src, OutputArray dst, float accuracy) {
+  /**
+   * Implementation of the Radon Transform
+   *
+   * Example:
+   * 					::
+   * 					cv::Mat img, imgRadTrans;
+   * 					:::: img holds some image ::::
+   *
+   * 					radonTransform(img, imgRadTrans, 0.5);
+   * 					:::: imgRadTrans contains the radon transform ::::
+   */
+  // Asserting accuracy to be positive
+  CV_Assert(accuracy > 0);
+
+  Mat srcMat_ = src.getMat(), srcMat;
+  CV_Assert(srcMat_.type() == CV_8UC1);
+
+  srcMat_.convertTo(srcMat, CV_32F, 1.0/255);
+
+  int rows = srcMat.rows, cols = srcMat.cols;
+  float diag = (float)std::sqrt(rows*rows + cols*cols);
+
+  Mat srcMatPadded;
+  int padWidth = (int)((std::ceil(diag - cols) + 2)/2);
+  int padHeight = (int)((std::ceil(diag - rows) + 2)/2);
+  copyMakeBorder(srcMat, srcMatPadded, padWidth, padHeight, padWidth, padHeight, BORDER_CONSTANT, 0);
+
+  int n = (int)(180/accuracy) + 1;
+  std::vector<float> vecTheta = linspace(0, 180, n);
+  int dstCols = static_cast<int>(vecTheta.size()), dstRows = srcMatPadded.rows;
+
+  dst.create(Size(dstCols, dstRows), CV_32F);
+  Mat dstMat = dst.getMat();
+
+  int num = srcMatPadded.cols;
+  std::vector<float> vecVal = linspace(-1, 1, num);
+
+  Mat meshX, meshY, meshXX, meshYY;
+  meshgrid(Mat(vecVal), Mat(vecVal), meshX, meshY);
+
+  for(int i = 0; i < dstCols; ++i) {
+    float theta = (90 - vecTheta[i])*static_cast<float>(CV_PI)/180.0f;
+    float cosTh = std::cos(theta), sinTh = std::sin(theta);
+
+    Mat temp(srcMatPadded.size(), CV_32F);
+
+    // Rotating meshgrid entries by an angle theta
+    meshXX = cosTh*meshX - sinTh*meshY;
+    meshYY = sinTh*meshX + cosTh*meshY;
+
+    // Clamping values to [-1, 1]
+    for(int l = 0; l < meshXX.cols; ++l) {
```

Comment:
90.0f ?

> +  for(int i = 0; i < dstCols; ++i) {
> +    float theta = (>>>>90.0f<<<<< - vecTheta[i])*static_cast<float>(CV_PI)/180.0f;
> +    float cosTh = std::cos(theta), sinTh = std::sin(theta);



On 13/02/2016 11:59, Balint Cristian wrote:
> > +  for(int i = 0; i < dstCols; ++i) {
> > +    float theta = (90 - vecTheta[i])*static_cast<float>(CV_PI)/180.0f;
> > +    float cosTh = std::cos(theta), sinTh = std::sin(theta);

-- 
Best regards
Rob Colclough
CTO
Sentel Advance - OwnTV Group
T. US +1 215 240 6030
T. UK +44 (0)20 8144 3150 (Low-rate 03333 442 120)
www.Sentel-Advance.com
Skype: ViewPoint3D
YouTube: www.youtube.com/viewpoint3d
  
ViewPoint 3D software offers a fresh and immediate way to create stunning 3D presentations, info TV, content and simulations with real-time data for almost any type of application.

This email and any files transmitted with it are confidential and intended solely for the use of the individual
or entity to whom they are addressed. If you have received this email in error please notify the system manager.
This message contains confidential information and is intended only for the individual named. If you are not the
named addressee you should not disseminate, distribute or copy this e-mail. Please notify the sender immediately
by e-mail if you have received this e-mail by mistake and delete this e-mail from your system. If you are not the
intended recipient you are notified that disclosing, copying, distributing or taking any action in reliance on the
contents of this information is strictly prohibited.